src/editor/Preview.js

```diff
@@ -5,35 +5,49 @@ import { Button } from '@wordpress/components';
 import { withSelect, withDispatch } from '@wordpress/data';
 import { compose } from '@wordpress/compose';
 import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
 
 /**
  * External dependencies
  */
-import { stringify } from 'qs';
 import { WebPreview } from 'newspack-components';
 
 const PreviewSetting = ( { autosavePost, isSavingPost, postId, metaFields } ) => {
 	const previewQueryKeys = window.newspack_popups_data?.preview_query_keys || {};
+	const frontendUrl = window?.newspack_popups_data?.frontend_url || '/';
+	const onLoad = () => {};
 	const abbreviatedKeys = {};
 	Object.keys( metaFields ).forEach( key => {
 		if ( previewQueryKeys.hasOwnProperty( key ) ) {
 			abbreviatedKeys[ previewQueryKeys[ key ] ] = metaFields[ key ];
 		}
 	} );
 
-	const query = stringify( {
+	const query = {
 		pid: postId,
 		// Autosave does not handle meta fields, so these will be passed in the URL
 		...abbreviatedKeys,
-	} );
+	};
 
 	const isArchivePagesPrompt = metaFields.placement === 'archives';
 	const previewURL =
 		window.newspack_popups_data[ isArchivePagesPrompt ? 'preview_archive' : 'preview_post' ] || '/';
 
+	const onWebPreviewLoad = iframeEl => {
+		if ( iframeEl ) {
+			[
+				...iframeEl.contentWindow.document.querySelectorAll( 'a[href^="' + frontendUrl + '"]' ),
+			].forEach( anchor => {
+				anchor.setAttribute( 'href', addQueryArgs( anchor.getAttribute( 'href' ), query ) );
+			} );
+			onLoad( iframeEl );
```

Comment:
On line 18 `onLoad` is defined as an "empty" function. What purpose does it serve?

Reply:
Ah yes, that's not needed. I was following the example of the `SegmentationPreview` component and forgot to remove this piece (which is used in that component but passed as a prop).